### ISSUE TITLE
Switch to int64-buffer

### ISSUE THREAD

**Issue by kolodny** [googleapis/nodejs-bigtable#94]
We currently use `node-int64` which is slower and less correct than `int64-buffer`:

```js
const Int64BE = require('int64-buffer').Int64BE;
const Int64 = require('node-int64');

const test = (number) => {
  const int64 = new Int64(number);
  const int64BE = new Int64BE(number);
  console.log('int64', int64.toNumber())
  console.log('int64BE', int64BE.toNumber())
  console.log({
    MAX_SAFE_INTEGER: Number.MAX_SAFE_INTEGER,
    isNumberTooBig:  int64BE.toNumber() > Number.MAX_SAFE_INTEGER,
  })
  console.log('int64', int64.toString())
  console.log('int64 Matches', int64.toString() === number)
  console.log('int64BE', int64BE.toString())
  console.log('int64BE Matches', int64BE.toString() === number)
  console.log('equals', int64.toBuffer().equals(int64BE.toBuffer()))
}

const a = number => new Int64(number).toBuffer();
const b = number => (new Int64BE(number)).toBuffer();

console.time('Int64');
for (var i = 0; i < 100000; i++) {
  a(123)
}
console.timeEnd('Int64');

console.time('Int64BE');
for (var i = 0; i < 100000; i++) {
  b(123)
}
console.timeEnd('Int64BE');

test('1234567890123456789')
test('9007199254740991')
```

outputs:

```
Int64: 162.114ms
Int64BE: 20.978ms
int64 Infinity
int64BE 1234567890123456800
{ MAX_SAFE_INTEGER: 9007199254740991, isNumberTooBig: true }
int64 Infinity
int64 Matches false
int64BE 1234567890123456789
int64BE Matches true
equals false
int64 -Infinity
int64BE 9007199254740991
{ MAX_SAFE_INTEGER: 9007199254740991, isNumberTooBig: false }
int64 -Infinity
int64 Matches false
int64BE 9007199254740991
int64BE Matches true
equals false
```

**Comment by ajaaym** [comment]
@kolodny any idea about [long.js from decodeIO](https://github.com/dcodeIO/long.js) which is used in protobufjs as well?

**Comment by kolodny** [comment]
Yup, looks like it's even faster

```
Int64: 180.108ms
Int64BE: 25.671ms
long: 3.037ms
```

**Comment by vijay-qlogic** [comment]
@kolodny 
I could not find `toBuffer` method in `long.js` as available in `node-int64` and `int64-buffer`
Is there any to perform `toBuffer` using Long.js? 
Or we shall use `int64-buffer`?

**Comment by kolodny** [comment]
It has a `toBytesBE()` on the object which gives an array of bits. You can use `Buffer.from` to convert that to a buffer:

```js
const Long = require('long');
const Int64BE = require('int64-buffer').Int64BE;
const Int64 = require('node-int64');

const a = number => new Int64(number).toBuffer();
const b = number => (new Int64BE(number)).toBuffer();
const c = number => Buffer.from(Long.fromNumber(number).toBytesBE())
```

**Comment by vijay-qlogic** [comment]
Okay, I tried methods a, b, c on same number but output differs for Long.js

> a(9999999999999999999)
<Buffer 8a c7 23 04 89 e8 00 00>

> b(9999999999999999999)
<Buffer 8a c7 23 04 89 e8 00 00>

> c(9999999999999999999)
<Buffer 7f ff ff ff ff ff ff ff>

**Comment by kolodny** [comment]
I wouldn't worry too much about large numbers, all the options start having non senible values at that point

```js
9999999999999999999 // 10000000000000000000
new Int64(9999999999999999999).toNumber() // -Infinity
new Int64('9999999999999999999').toNumber() // -Infinity

Int64BE("9999999999999999999").toNumber(); // -8446744073709552000
Int64BE("9999999999999999999").toString(); // "-8446744073709551617"
Int64BE(9999999999999999999).toString(); // "-8446744073709551616"
Int64BE(9999999999999999999).toNumber(); // -8446744073709552000

Long.fromValue(9999999999999999999).toNumber(); // 9223372036854776000
Long.fromValue('9999999999999999999').toNumber(); // -8446744073709552000
```

Anyone working with numbers that large should know that javascript can't even express numbers that massive `9999999999999999999 > Number.MAX_SAFE_INTEGER` so they should expect to do some manual buffer managing and/or pull in something like [substack/node-bigint](https://github.com/substack/node-bigint)

**Comment by kolodny** [comment]
It's possible we can do something smarter once https://github.com/tc39/proposal-bigint makes its way into node, but until then we have to let the user figure this one out. It may be worth putting something in the docs if this becomes something that people are running into

**Comment by vijay-qlogic** [comment]
Ok, got your point, I'll use Long.js and do the enhancement

**Comment by kolodny** [comment]
Fixed in #156
